src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -192,25 +223,19 @@ private Course tryGetCourse(@NotNull Project project, @NotNull URL courseUrl) {
     try {
       return courseFactory.fromUrl(courseUrl, project);
     } catch (IOException e) {
-      notifyNetworkError();
+      notifier.notify(new NetworkErrorNotification(e), project);
       return null;
     } catch (MalformedCourseConfigurationFileException e) {
       logger.error("Malformed course configuration file", e);
-      notifyMalformedCourseConfiguration();
+      notifier.notify(new CourseConfigurationError(e), project);
       return null;
     }
   }
 
-  private void startAutoInstallsWithRestart(@NotNull Course course,
-                                            boolean restartWhenFinished,
-                                            @NotNull Project project) {
+  private Awaitable startAutoInstallsWithRestart(@NotNull Course course, @NotNull Project project) {
```

Comment:
perhaps rename this method to `startAutoInstalls` after this change?

Reply:
I'll keep that in mind for the future.